src/Java.Interop.Tools.JavaCallableWrappers/Java.Interop.Tools.JavaCallableWrappers/Crc64.cs

```diff
@@ -184,9 +55,30 @@ public override void Initialize ()
 
 		protected override void HashCore (byte [] array, int ibStart, int cbSize)
 		{
-			for (int i = ibStart; i < cbSize; i++) {
-				crc = Table [(byte) (crc ^ array [i])] ^ (crc >> 8);
+			int len = cbSize;
+			int idx = ibStart;
+
+			while (len >= 8) {
+				crc ^= (ulong)array[idx];
+				crc =
+					Table [7, crc & 0xff] ^
+					Table [6, (crc >> 8) & 0xff] ^
+					Table [5, (crc >> 16) & 0xff] ^
+					Table [4, (crc >> 24) & 0xff] ^
+					Table [3, (crc >> 32) & 0xff] ^
+					Table [2, (crc >> 40) & 0xff] ^
+					Table [1, (crc >> 48) & 0xff] ^
+					Table [0, crc >> 56];
```

Comment:
Would it help even further to use `unsafe` for accessing this array?

Reply:
It did, it made the new code 5.6 times faster